tests/platform_tests/test_reboot.py

```diff
@@ -87,22 +87,31 @@ def reboot_and_check(localhost, dut, interfaces, xcvr_skip_list,
     # Append the last reboot type to the queue
     logging.info("Append the latest reboot type to the queue")
     REBOOT_TYPE_HISTOYR_QUEUE.append(reboot_type)
-
+    # wait for dut to come back before checking critical processes/intfs
+    wait_for_startup(dut, localhost, delay=10, timeout=800)
     check_interfaces_and_services(dut, interfaces, xcvr_skip_list, reboot_type=reboot_type)
     if dut.is_supervisor_node():
         for lc in duthosts.frontend_nodes:
             wait_for_startup(lc, localhost, delay=10, timeout=300)
             check_interfaces_and_services(lc, interfaces, xcvr_skip_list)
 
 
-def check_interfaces_and_services(dut, interfaces, xcvr_skip_list,
+def check_interfaces_and_services(dut, interfaces, xcvr_skip_list, localhost,
                                   interfaces_wait_time=MAX_WAIT_TIME_FOR_INTERFACES, reboot_type=None):
     """
     Perform a further check after reboot-cause, including transceiver status, interface status
     @param localhost: The Localhost object.
     @param dut: The AnsibleHost object of DUT.
     @param interfaces: DUT's interfaces defined by minigraph
     """
+    # Here we wait for device to boot up if reboot_type is known, in this case we know
+    # it's checking intfs and services after reboot.
+    # In other cases, if caller is only shutting down some services, or, it's in some test teardown
+    # we need to call wait_for_startup before calling this function.
+    logging.info("wait for device to boot up for {}sec when reboot_type is known"
```

Comment:
this log should move under the conditional check for reboot_type as this is done conditionally only if this function is called after reboot is performed...